src/settings/components/SettingBlocks.tsx

```diff
@@ -411,41 +510,59 @@ const ModelSettingsComponent: React.FC<ModelSettingsComponentProps> = ({
-              options={providers}
-              value={newModel.provider}
-              onChange={(value) => {
-                setNewModel({ ...newModel, provider: value });
-              }}
-            />
-            <TextComponent
-              name="Base URL (optional)"
-              description="For 3rd party OpenAI Format endpoints only. Leave blank for other providers."
-              value={newModel.baseUrl || ""}
-              placeholder="https://api.example.com/v1"
-              onChange={(value) => setNewModel({ ...newModel, baseUrl: value })}
-            />
-            <TextComponent
-              name="API Key (optional)"
-              description="API key for the 3rd party provider"
-              value={newModel.apiKey || ""}
-              placeholder="Enter API key"
-              type="password"
-              onChange={(value) => setNewModel({ ...newModel, apiKey: value })}
-            />
-            <button onClick={handleAddModel} className="add-model-button">
-              Add Model
-            </button>
+            {currentProviderOptions.model && (
+              <TextComponent
+                name="Model Name"
+                description={`The name of the model, i.e. ${isEmbeddingModel ? "text-embedding-3-small" : "gpt-4o-mini"}`}
+                value={newModel.name}
+                placeholder="Enter model name"
+                onChange={(value) => {
+                  setNewModel({ ...newModel, name: value });
+                }}
+              />
+            )}
+
+            {currentProviderOptions.provider && (
+              <DropdownComponent
+                name="Provider"
+                options={providers}
+                value={newModel.provider}
+                onChange={(value) => {
+                  setNewModel({ ...newModel, provider: value });
+                }}
+              />
+            )}
+
+            {currentProviderOptions.baseUrl && (
+              <TextComponent
+                name="Base URL (optional)"
+                description="If you are using a proxy, enter the proxy URL here. Otherwise, leave it blank."
+                value={newModel.baseUrl || ""}
+                placeholder="https://api.example.com/v1"
+                onChange={(value) => setNewModel({ ...newModel, baseUrl: value })}
+              />
+            )}
+
+            {currentProviderOptions.apiKey && (
+              <TextComponent
+                name="API Key (optional)"
+                description="Please fill in the key, unless it is not required."
+                value={newModel.apiKey || ""}
+                placeholder="Enter API key"
+                type="password"
+                onChange={(value) => setNewModel({ ...newModel, apiKey: value })}
+              />
+            )}
+            <div>
+              <button onClick={handleAddModel} className="add-model-button">
+                Add Model
+              </button>
+              {!isEmbeddingModel && (
```

Comment:
Why not have the verify button for embedding models?

Reply:
forget. I will add it later.